src/mem/allocconfig.h

```diff
@@ -4,6 +4,34 @@
 
 namespace snmalloc
```

Comment:
I wonder if we should perhaps have a pal_pre_include.h that contains the enums required to define a pal and nothing else.

Reply:
Shall I create pal_consts.h?

Reply:
That sounds good. Thanks